### DISCUSSION TITLE
A tutorial for custom Android code e.g. HorizontalPager?

### DISCUSSION THREAD

**Discussion by sturmenta** [skiptools/skip#240] [General]
Hello! I just start with Skip and I want to implement an horizontal slides for the onboarding of my app

https://github.com/user-attachments/assets/57aed2a0-57e3-4db3-adb2-713367f88762

I can make it work on iOS using TabView, but is not so easy for android

```swift
        TabView(selection: $selection) {
            ...
        }
        .tabViewStyle(.page(indexDisplayMode: .always))
        .background(AppColor.appPrimaryColor)
        .animation(Animation.easeInOut(duration: 1.0), value: selection)
        .transition(.slide)
```

I found that someone already need something like this [Skip Discussion - Implementation of .page for tabViewStyle](https://github.com/orgs/skiptools/discussions/137)

but I can't make it work for the android side, I think maybe a video tutorial for a case like this with the extra steps for android configuration or extra documentation would be really great

Thanks!

**Comment by aabewhite** [comment]
Thanks for the suggestion. You're right - unfortunately we don't have any control like this yet. If you're OK with diving into Compose, my suggestion would be:

1. Create a scratch project in Android Studio and implement this in Compose.
2. Drop the .kt file with your implementation into your Skip project: https://skip.tools/docs/platformcustomization/#kotlin-files
3. Call your Compose implementation via ComposeView: https://skip.tools/docs/modules/skip-ui/#composeview

Please let us know how it goes!
